system/Database/BaseBuilder.php

```diff
@@ -876,7 +874,7 @@ protected function _whereIn(?string $key = null, $values = null, bool $not = fal
         $prefix = empty($this->{$clause}) ? $this->groupGetType('') : $this->groupGetType($type);
 
         $whereIn = [
-            'condition' => $prefix . $key . $not . ($values instanceof Closure ? " IN ({$ok})" : " IN :{$ok}:"),
+            'condition' => $prefix . $key . $not . ($this->isSubquery($values) ? " IN ({$ok})" : " IN :{$ok}:"),
```

Comment:
Could we not spare the inline-if by passing true to the buildSubquery call as second parameter?

Reply:
Let's try.